src/seedu/addressbook/Main.java

```diff
@@ -107,6 +107,9 @@ private CommandResult executeCommand(Command command)  {
         try {
             command.setData(addressBook, lastShownList);
             CommandResult result = command.execute();
+            if(!storage.isFileExist()){
+            	ui.showFileNotFound();
+            }
```

Comment:
This may be a too high-level method to check for this low-level problem. StorageFile::save() may be a more suitable place to detect it.